### PR TITLE
Tradeable cashflow hardhat

### PR DESCRIPTION
**Pull request by saflamini**
1) Added a new example called tradeabe-cashflow hardhat as an example which makes use of the sdk core for testing, and leverages hardhat for one of the most often used Superfluid examples
2) Made fixes to tests which were broken in the streaming-call-option example
3) Fixed a broken link in the examples readme

NOTE: there is a development environment issue inside of the NFT billboard example which is causing one of the checks to fail for it. Running yarn test and the build-and-test.sh script seems to work fine. I will continue to investigate, but the due to the tests passing & build working properly when run locally, I will choose to fix this down the line.

@hellwolf I intend to use this repo during a workshop this week as well.

-Sam

### CODE REVIEW

**Comment by saflamini** [comment]
@0xdavinchee what did you think here?

The NFTbillboard example is failing without lockfile, but works otherwise. I plan to investigate and fix that early next week. The new hardhat/tcf example is here though

**Comment by 0xdavinchee** [comment]
will take a closer look tomorrow, but I would add `artifacts` into your `.gitignore`. it's good practice to gitignore generated files unless they're necessary.

**Comment by saflamini** [comment]
> will take a closer look tomorrow, but I would add `artifacts` into your `.gitignore`. it's good practice to gitignore generated files unless they're necessary.

Done

**Comment by 0xdavinchee** [comment]
> > will take a closer look tomorrow, but I would add `artifacts` into your `.gitignore`. it's good practice to gitignore generated files unless they're necessary.
> 
> Done

awesome, I should've mentioned this yesterday as well, but I am gonna delete the artifacts folder and commit. I forget that if you've already committed, adding to gitignore doesn't remove the files

**Review by hellwolf — APPROVED** [review]
It seems I'm the code owner of .github/workflows/handler.build-examples.yml, I will approve it as it is. I have not tested the nftbillboard-userdata example though, but we can merge too if example build  passes.

File: examples/tradeable-cashflow-hardhat/contracts/RedirectAll.sol [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            !_host.isApp(ISuperApp(newReceiver)),
+            "New receiver can not be a superApp"
+        );
+        if (newReceiver == _receiver) return;
+        // @dev delete flow to old receiver
+        (, int96 outFlowRate, , ) = _cfa.getFlow(
+            _acceptedToken,
+            address(this),
+            _receiver
+        ); //CHECK: unclear what happens if flow doesn't exist.
```
Comment (0xdavinchee) [inline examples/tradeable-cashflow-hardhat/contracts/RedirectAll.sol:140]: I'm just gonna assume this was checked, looks like a copy pasta from the OG `RedirectAll.sol`

Comment (saflamini) [reply examples/tradeable-cashflow-hardhat/contracts/RedirectAll.sol:140]: Yes copy pasta.

**Pull request merged**